lib/jobs/secure-erase-drive.js

```diff
@@ -439,5 +440,57 @@ function secureEraseJobFactory(
         };
     };
 
+    /**
+     * @memberOf SecureEraseJob
+     * @function _classifyDiskPara
+     * @description classify disk informations inside eraseSettings into devName list and wwid list
+     */
+    SecureEraseJob.prototype._classifyDiskPara = function() {
+        var eraseSettings = this.eraseSettings;
+        _.forEach(eraseSettings, function(eraseSetting){
+            var diskList = [];
+            var diskWwid = {};
+            _.forEach(eraseSetting.disks, function(disk){
+                //To make sure rackhd drive catalog matches real situation on nodes
+                //user is asked to input both devname and esxiwwid for each disk
+                if (disk.devName) {
+                    diskList.push(disk.devName);
+                } else {
+                    throw new Error("No devName is found from user input disk list");
+                }
+
+                if (disk.esxiWwid){
+                    diskWwid[disk.devName] = disk.esxiWwid;
+                } else{
+                    throw new Error("No esxiWwid is found from user input disk list");
+                }
+            });
+            eraseSetting.disks = diskList;
+            eraseSetting.wwids = diskWwid;
+        });
+    };
+
+    /**
+     * @memberOf SecureEraseJob
+     * @function _validateDiskWwid
+     * @description validate extended disk esxi wwid information
+     */
+    SecureEraseJob.prototype._validateDiskWwid= function(diskInfo) {
+        var wwidInfo = {};
+        var eraseSettings = this.eraseSettings;
+        _.forEach(eraseSettings, function(eraseSetting){
+            wwidInfo = _.defaults(wwidInfo, eraseSetting.wwids);
+            //wwid is only used for catalog matching, delete them after retrieving
+            delete eraseSetting.wwids;
+        });
+
+        _.forEach(diskInfo, function(disk){
+            if (wwidInfo[disk.devName] !== disk.esxiWwid) {
+                throw new Error("Drive id catalog does not match user input data, " +
+                                "drive re-cataloging is required");
```

Comment:
Will it be useful if adding the unmatched drive info (like devName and wwid) into the error message?